### PR TITLE
[C API] support string size for char pointer

### PR DESCRIPTION
**Pull request by riverlijunjie**
### Details:
 - Add string size for C API reading model from memory

### Tickets:
 - Closes https://github.com/openvinotoolkit/openvino/issues/19833

### CODE REVIEW

File: src/bindings/c/include/openvino/c/ov_core.h [hunk 1]
```diff
@@ -189,6 +189,28 @@ ov_core_read_model_from_memory(const ov_core_t* core,
...
+ * @param weights Shared pointer to a constant tensor with weights.
+ * @param model A pointer to the newly created model.
+ * Reading ONNX / PDPD / TF / TFLite models does not support loading weights from the @p weights tensors.
+ * @note Created model object shares the weights with the @p weights object.
+ * Thus, do not create @p weights on temporary data that can be freed later, since the model
+ * constant data will point to an invalid memory.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_read_model_from_memory_with_size(const ov_core_t* core,
```
Comment (ilyachur) [inline src/bindings/c/include/openvino/c/ov_core.h:208]: Maybe we can come up the better name... For example: `ov_core_read_model_from_memory_buffer`

Comment (ilyachur) [reply src/bindings/c/include/openvino/c/ov_core.h:208]: Also we can deprecate old method, because these methods provide the same functionality

Comment (riverlijunjie) [reply src/bindings/c/include/openvino/c/ov_core.h]: But if deprecate old method maybe lead to some compatibility problem if this API has been used by user application.

Comment (ilyachur) [reply src/bindings/c/include/openvino/c/ov_core.h]: Yes in this case user will get a deprecation warning, but it is a normal process and without it we cannot change or remove legacy API. I don't talk about deletion of legacy method immediately, only about the deprecation and removing later.
From my point of view two methods will confuse users...

**Review by ilyachur — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**